contracts/test/KyberElasticSampler.t.sol

```diff
@@ -0,0 +1,223 @@
+            console.log(
+                "Reverse Path Sell: test=%d, kyberQuoterGasUsage=%d, multiQuoterGasUsage=%d",
+                i + 1,
+                kyberQuoterGasUsage,
+                multiQuoterGasUsage
+            );
+            (kyberQuoterGasUsage, multiQuoterGasUsage) = compareQuoterBuys(path, testAmounts[i]);
+            console.log(
+                "Normal Path Buy: test=%d, kyberQuoterGasUsage=%d, multiQuoterGasUsage=%d",
+                i + 1,
+                kyberQuoterGasUsage,
+                multiQuoterGasUsage
+            );
+            (kyberQuoterGasUsage, multiQuoterGasUsage) = compareQuoterBuys(reversePath, testAmounts[i]);
+            console.log(
+                "Reverse Path Buy: test=%d, kyberQuoterGasUsage=%d, multiQuoterGasUsage=%d",
+                i + 1,
+                kyberQuoterGasUsage,
+                multiQuoterGasUsage
+            );
+        }
+    }
+
+    function compareQuoterSells(
+        bytes memory path,
+        uint256[] memory amountsIn
+    ) private returns (uint256 kyberQuoterGasUsage, uint256 multiQuoterGasUsage) {
+        uint256 gas0 = gasleft();
+        (uint256[] memory multiQuoterAmountsOut, ) = multiQuoter.quoteExactMultiInput(factory, path, amountsIn);
+        uint256 gas1 = gasleft();
+
+        for (uint256 i = 0; i < amountsIn.length; ++i) {
+            try kyberQuoter.quoteExactInput(path, amountsIn[i]) returns (
+                uint256 kyberQuoterAmountOut,
+                uint160[] memory /* sqrtPriceX96AfterList */,
+                uint32[] memory /* initializedTicksCrossedList */,
+                uint256 /* gasEstimate */
+            ) {
+                assertApproxEqRel(multiQuoterAmountsOut[i], kyberQuoterAmountOut, ERROR_THRESHOLD_10_BPS);
+            } catch {}
+        }
+        return (gas1 - gasleft(), gas0 - gas1);
+    }
+
+    function compareQuoterBuys(
+        bytes memory path,
+        uint256[] memory amountsOut
+    ) private returns (uint256 kyberQuoterGasUsage, uint256 multiQuoterGasUsage) {
+        uint256 gas0 = gasleft();
+        (uint256[] memory multiQuoterAmountsIn, ) = multiQuoter.quoteExactMultiOutput(factory, path, amountsOut);
+        uint256 gas1 = gasleft();
+
+        for (uint256 i = 0; i < amountsOut.length; ++i) {
+            try kyberQuoter.quoteExactOutput(path, amountsOut[i]) returns (
+                uint256 kyberQuoterAmountIn,
+                uint160[] memory /* sqrtPriceX96AfterList */,
+                uint32[] memory /* initializedTicksCrossedList */,
+                uint256 /* gasEstimate */
+            ) {
+                assertApproxEqRel(multiQuoterAmountsIn[i], kyberQuoterAmountIn, ERROR_THRESHOLD_10_BPS);
+            } catch {}
+        }
+        return (gas1 - gasleft(), gas0 - gas1);
+    }
+
+    // TODO: the helper functions below are very similar to those in UniswapV3Common, refactor.
```

Comment:
Yep, think we should refactor. I'm just not sure how we would be able to use one function when UniswapV3 and KyberElastic pools have separate interfaces for fees. Maybe something like this where we pass the signature of the fee method?
```solidity
function toPath(address[] memory tokenPath, address[] memory poolPath, bytes4 feeMethodId) internal view returns (bytes memory path) {
...
uint24 poolFee = poolPath[i-1].call(feeMethodId);
...
```

Reply:
Yeah or some kind of facade wrapper contract that provides the same interface for each pool or DEX